probers/aws/register.go

```diff
@@ -0,0 +1,86 @@
+package aws
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/Symantec/tricorder/go/tricorder"
+	"github.com/Symantec/tricorder/go/tricorder/units"
+	"io"
+	"net/http"
+	"strings"
+)
+
+var versionFilename string = "/proc/version"
+
+var (
+	metaDataKeyToDescription = map[string]string{
+		"ami-id":          "AMI ID",
+		"hostname":        "local hostname",
+		"instance-id":     "AWS instance ID",
+		"instance-type":   "AWS instance type",
+		"local-ipv4":      "local ipv4",
+		"public-hostname": "public hostname",
+		"public-ipv4":     "public ipv4",
+	}
+)
+
+func register(dir *tricorder.DirectorySpec) *prober {
+	if keys, err := getMetaDataKeys(); err == nil {
+		for _, key := range keys {
+			if desc, ok := metaDataKeyToDescription[key]; ok {
+				if value, err := getByMetaDataKey(key); err == nil {
+					if err := dir.RegisterMetric(
+						key,
+						&value,
+						units.None,
+						desc); err != nil {
+						panic(err)
+					}
+				}
+			}
+		}
+	}
+	p := new(prober)
+	return p
+}
+
+func getMetaDataKeys() ([]string, error) {
+	resp, err := http.Get("http://169.254.169.254/latest/meta-data/")
```

Comment:
How can you tell that the metadata service is for an AWS instance rather than a metadata service for some other Cloud provider (i.e. OpenStack)?